src/crypto.rs

```diff
@@ -1,15 +1,16 @@
 use crate::bytes::write_bytes_to_lower_hex;
+use crate::prelude::*;
 use crate::primitives::Bytes32;
 #[cfg(feature = "serde")]
 use crate::serde::{try_bytes_from_hex_str, HexError};
 use crate::ssz::ByteVector;
-use blst::{min_pk as bls_impl, BLST_ERROR};
+use error_chain::ChainedError;
+use milagro_bls::{AggregatePublicKey, AggregateSignature, AmclError};
```

Comment:
this may be its own sidequest but I would strongly prefer to use `blst`

do you know why it is not `no-std` compatible?

Reply:
According to the `blst` crate, it is not no-std compatible hence the usage of `milagro_bls` crate

Reply:
blst crate is only supports compilation to `wasm32-unknown-emscripten` because of it's dependence on some FFIs, but `wasm32-unknown-unknown` is the target we'll need in the future.
More details here https://github.com/rustwasm/team/issues/291